### ISSUE TITLE
Lint does not pick up exclamation mark (!) for negation in shell as shell functionality (ANSIBLE0013)

### ISSUE THREAD

**Issue by Rickkwa** [ansible/ansible-lint#363]
# Issue Type
- Bug report

# Ansible and Ansible Lint details

```
ansible 2.4.0.0
  config file = /root/.ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Nov  6 2016, 00:28:07) [GCC 4.8.5 20150623 (Red Hat 4.8.5-11)]

ansible-lint 3.4.23
```

- ansible installation method: pip
- ansible-lint installation method: pip

# Desired Behaviour

If you run the lint on a `shell` task that uses `!` for returncode negation, the desired behavior is to NOT return `[ANSIBLE0013] Use shell only when shell functionality is required`.

# Actual Behaviour (Bug report only)

The following sample playbook is reporting a linting error `ANSIBLE0013`.
```yaml
- hosts: localhost
  tasks:
    - name: test
      shell: "! grep 'ASD' /etc/ssh/sshd_config"
      changed_when: false
      register: foo
```

```bash
$ ansible-lint -p playbook.yml
test-general.yml:3: [EANSIBLE0013] Use shell only when shell functionality is required
```

Meaning, it wants me to use `command` instead of `shell` since it doesn't think that `!` is shell behavior. However, if I use `command`, the task fails with `[Errno 2] No such file or directory` which is obviously false as the file does indeed exist.
